### PR TITLE
fix: backend error handling

### PR DESCRIPTION
**Pull request by anusha-kommu**

### CODE REVIEW

File: generator/overrides/education/pages/issuer/credential-form/index.tsx [hunk 1]
```diff
@@ -16,7 +17,11 @@ const CredentialForm: FC = () => {
 
   useEffect(() => {
     if (error) {
-      showErrorToast(new Error(messages.issuer.error.apiError))
+      if (error.response?.data?.error?.code === ErrorCodes.INTERNAL_SERVER_ERROR) {
+        showErrorToast(new Error(messages.issuer.error.apiError))
+        return
+      }
+      showErrorToast(error)
```
Comment (OleksiiO-affinidi) [inline generator/overrides/education/pages/issuer/credential-form/index.tsx:24]:
```suggestion
      if (error.response?.data?.error?.code === ErrorCodes.INTERNAL_SERVER_ERROR) {
        showErrorToast(new Error(messages.issuer.error.apiError))
      } else {
        showErrorToast(error)
      }
```

File: generator/template/pages/api/holder/confirm-sign-in.ts [hunk 2]
```diff
@@ -22,12 +26,22 @@ async function handler(
...
+  try {
+    const { accessToken } = await cloudWalletClient.confirmSignInPasswordless({
+      token,
+      confirmationCode,
+    })
 
-  res.status(200).json({ accessToken })
+    res.status(200).json({ accessToken })
+  } catch (error: any) {
+    if (['COR-5', 'COR-17'].includes(error.response?.data?.code)) {
```
Comment (OleksiiO-affinidi) [inline generator/template/pages/api/holder/confirm-sign-in.ts:37]: pls move this code errors to enum

File: generator/template/pages/api/holder/get-vcs.ts [hunk 3]
```diff
@@ -1,28 +1,43 @@
...
+  try {
+    const { vcs } = await cloudWalletClient.getCredentials({}, { accessToken })
 
-  res.status(200).json({ vcs: vcs.filter(vc => vc.type.includes(schemaType)) })
+    const { schemaType } = parseSchemaURL(JSONLD_CONTEXT_URL)
+    res
+      .status(200)
+      .json({ vcs: vcs.filter((vc) => vc.type.includes(schemaType)) })
+  } catch (error: any) {
+    if (error.response?.data?.code === 'CWA-4') {
```
Comment (OleksiiO-affinidi) [inline generator/template/pages/api/holder/get-vcs.ts:33]: the same

File: generator/template/pages/api/middlewares/error-handler.ts [hunk 4]
```diff
@@ -11,24 +14,26 @@ export const errorHandler: Middleware = async (req, res, next) => {
...
         error: {
-          code: 'VALIDATION_ERROR',
+          code: ErrorCodes.VALIDATION_ERROR,
           issues: error.issues,
-        }
+        },
       })
     } else if (error instanceof ApiError) {
       logger.debug({ error }, 'API error')
+      console.log('isApi')
```
Comment (OleksiiO-affinidi) [inline generator/template/pages/api/middlewares/error-handler.ts:23]: remove pls

File: generator/overrides/education/pages/issuer/credential-form/index.tsx (hunk 1 above)
Comment (OleksiiO-affinidi) [reply generator/overrides/education/pages/issuer/credential-form/index.tsx:24]: and all other places with the same coditions

Comment (vitalyrudenko-affinidi) [reply generator/overrides/education/pages/issuer/credential-form/index.tsx:24]: Usually, we don't need to explicitly check that code equals INTERNAL_SERVER_ERROR. Usually this is reserved for cases when we couldn't handle the error:
```ts
if (error.code === 'ERROR_1') {

} else if (error.code === 'ERROR_2') {

} else {
  // assume it's internal server error or any unhandled error
}
```

I'm not sure what's the difference here between `showErrorToast(new Error(messages.issuer.error.apiError))` and `showErrorToast(error)`. Can't we just do this here?
```ts
if (error) {
  showErrorToast(new Error(messages.issuer.error.apiError))
}
```

File: generator/overrides/education/enums/errorCodes.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+export enum ErrorCodes {
```
Comment (OleksiiO-affinidi) [inline generator/overrides/education/enums/errorCodes.ts:1]: why not in the `template` folder?
they r equal for each app

Comment (anusha-kommu) [reply generator/overrides/education/enums/errorCodes.ts:1]: yes, they are equal. But we need ErrorCodes in `generator/overrides/education/pages/issuer/credential-form/index.tsx` for that had to create enum in overrides

Comment (OleksiiO-affinidi) [reply generator/overrides/education/enums/errorCodes.ts:1]: if u r added it to `template/enums/errorCodes` it will auto-generate it everywhere

**Review by OleksiiO-affinidi — APPROVED** [review]

**Pull request merged**